presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaMetadata.java

```diff
@@ -225,6 +227,11 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
 
         log.info("startTs: %s, endTs: %s", startTs, endTs);
+        if (config.isHardLimitOn() && startTs == null && endTs == null) {
```

Comment:
I don't understand how the hard_limit is applied. Can you add more comments to make it clear? Especially for the  `for (Map.Entry<ColumnHandle, Domain> entry : columnHandleDomainMap.entrySet())` loop above.

Also, could we check whether it meets the expectations:
1. If users specify a time range exceeding the limitation, an exception should be thrown directly. Otherwise, the behavior is changed silently and unexpectedly.
2. If users do not specify the range, we could either throw an exception directly, or trim to the limited time range with an **explicit logging/print-out** notifying users.